### PR TITLE
Gui Rodrigues - Scrabble Module 1

### PR DESCRIPTION
**Pull request by qruz-hq**

### CODE REVIEW

File: src/scrabble.js [hunk 1]
```diff
@@ -1,6 +1,55 @@
...
+  score () {
+    let s = 0
+    let doubleNext = false
+    let tripleNext = false
+    if (!this.isValid()) return s
+    this.word
+      .replace(' ', '')
+      .toLowerCase()
+      .split('')
+      .forEach((ch, index) => {
```
Comment (Anicaaa) [inline src/scrabble.js:27]: Interesting. I wonder what 'ch' does. Never used or seen it before.

File: src/scrabble.js [hunk 2]
```diff
@@ -47,8 +40,14 @@ class Scrabble {
         } else s += this.getCharScore(ch)
       })
 
-    if (this.word.indexOf('[') === 0 && this.word.indexOf(']') === this.word.length - 1) return s * 3
-    if (this.word.indexOf('{') === 0 && this.word.indexOf('}') === this.word.length - 1) return s * 2
+    if (
+      this.word.indexOf('[') === 0 &&
+      this.word.indexOf(']') === this.word.length - 1
```
Comment (Hoolio333) [inline src/scrabble.js:45]: Why word length - 1?

File: src/scrabble.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+    let s = 0
+    if (!this.isValid()) return s
+    this.word
+      .replace(' ', '')
+      .toLowerCase()
+      .split('')
+      .forEach((ch) => (s += this.getCharScore(ch)))
+
+    return s
+  }
```
Comment (Hoolio333) [inline src/scrabble.js:52]: I like this layout, very clean & clear - easy to read

File: src/scrabble.js [hunk 4]
```diff
@@ -1,6 +1,35 @@
+const ScoreUtil = require('./utils/Scores')
+
```
Comment (Hoolio333) [inline src/scrabble.js:2]: Not many pushes (I did the same), think we should probably push at every stage more.

File: src/utils/Scores.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  q: 10,
+  r: 1,
+  s: 1,
+  t: 1,
+  u: 1,
+  v: 4,
+  x: 8,
+  y: 4,
+  w: 4,
+  z: 10
```
Comment (Anicaaa) [inline src/utils/Scores.js:27]: Good to know there are other ways to code in Javascript. Here, I personally used the For Loop.

File: src/utils/Scores.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+module.exports = {
+  scores: scores
+}
```
Comment (Tibor22) [inline src/utils/Scores.js:32]: A good way to keep it modular

File: src/scrabble.js (hunk 2 above)
Comment (Creatorrr1) [reply src/scrabble.js:45]: I think that's for last index

File: src/scrabble.js [hunk 7]
```diff
@@ -12,8 +12,7 @@ class Scrabble {
   }
 
   getCharScore (char) {
-    if (ScoreUtil.scores[char]) return ScoreUtil.scores[char]
-    else return 0
+    return ScoreUtil.scores[char] || 0
```
Comment (Creatorrr1) [inline src/scrabble.js:15]: The or 0 was very efficient and line efficient, it even executed one of the test. Simple and clean code is the way 👌

**Pull request closed without merging**